lib/core/device_handler.rb

```diff
@@ -84,6 +84,8 @@ def create_single_device(case_role, role_app, case_caps)
       config_caps = convert_yaml(config_device["capabilities"])
       log_info("Role '#{case_role}': Adding capabilities from config: #{config_caps}")
       udid = config_caps["udid"] if config_caps.key?("udid")
+      @driver_port = config_caps["systemPort"].to_i if config_caps.key?("systemPort")
+      @server_port = config_caps["wdaLocalPort"].to_i if config_caps.key?("wdaLocalPort")
```

Comment:
This should still be @driver_port, it's just that Android and iOS have different names for their implementation. You can add another line for @server_port if you wish.